### PR TITLE
Feature/docker image deploy

### PR DESCRIPTION
**Pull request by enrique-alonso-martin-pw**
Now pushing to docker hub using `tcpmocker` account. It will only push the container and deploy the jar files when building the `master` branch.

### CODE REVIEW

**Comment by enrique-alonso-martin-pw** [comment]
As I have disable pushing on branches, you will only know if this works after merging. Not the best, but I did test it before that was introduced and it pushed the container to docker hub.

File: pom.xml [hunk 1]
```diff
@@ -89,6 +88,7 @@
         <!-- Maven -->
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.deploy.skip>true</maven.deploy.skip>
```
Comment (dmalch) [inline pom.xml]: Is it needed? It seems to be always provided from the outside.

File: tcp-mocker-app-test/src/test/java/io/payworks/labs/tcpmocker/TcpMockerAppDockerIT.java [hunk 2]
```diff
@@ -19,7 +19,10 @@
     private static final DockerComposeContainer tcpMocker =
             new DockerComposeContainer(new File("tcp-mocker-app/docker-compose.yml"))
                     .withExposedService(TCP_MOCKER_APP_SERVICE_NAME, TCP_SERVICE_PORT)
-                    .withExposedService(TCP_MOCKER_APP_SERVICE_NAME, WEB_SERVICE_PORT);
+                    .withExposedService(TCP_MOCKER_APP_SERVICE_NAME, WEB_SERVICE_PORT)
+                    .withEnv("TCP_MOCKER_APP_TAG", System.getenv("PROJECT_VERSION"))
```
Comment (dmalch) [inline tcp-mocker-app-test/src/test/java/io/payworks/labs/tcpmocker/TcpMockerAppDockerIT.java]: Will it also work on a local build?

File: pom.xml (hunk 1 above)
Comment (enrique-alonso-martin-pw) [reply pom.xml]: It is provided if we build it with travis. I wanted to make sure we don't deploy it by mistake if locally built.

File: tcp-mocker-app-test/src/test/java/io/payworks/labs/tcpmocker/TcpMockerAppDockerIT.java (hunk 2 above)
Comment (enrique-alonso-martin-pw) [reply tcp-mocker-app-test/src/test/java/io/payworks/labs/tcpmocker/TcpMockerAppDockerIT.java]: Good point. If not defined, there will be not tag and docker will default to latest, but I'll check and confirm it.

Comment (enrique-alonso-martin-pw) [reply tcp-mocker-app-test/src/test/java/io/payworks/labs/tcpmocker/TcpMockerAppDockerIT.java]: If the `PROJECT_VERSION` env variable is not found, it now falls back to `LOCAL-SNAPSHOT`

**Pull request merged**